nsqadmin/static/js/models/channel.js

```diff
@@ -14,24 +16,28 @@ var Channel = Backbone.Model.extend({
             encodeURIComponent(this.get('name')));
     },
 
-    initialize: function() {
-      this.on('change:clients', this.calculateHostnamePort);
-    },
+    parse: function(response) {
+        response['nodes'] = _.map(response['nodes'] || [], function(node) {
+            var nodeParts = node['node'].split(':');
+            node['show_broadcast_address'] = node['hostname'] !== nodeParts[0];
+            node['hostname_port'] = node['hostname'] + ':' + nodeParts[1];
+            return node;
+        });
+
+        response['clients'] = _.map(response['clients'] || [], function(client) {
+            var clientId = client['client_id'];
+            var hostname = client['hostname'];
 
-    calculateHostnamePort: function() {
-        this.get("clients").forEach(function(client){
-            var client_id = client["client_id"];
-            var hostname = client["hostname"];
-            var remote_address = client["remote_address"];
-  
             // ignore client_id if it's duplicative
-            if (client_id === hostname.split(".")[0] || client_id === hostname) {
-                client["client_id"] = "";
-            }
+            client['show_client_id'] = clientId !== hostname.split('.')[0] && clientId !== hostname;
+
+            var port = client['remote_address'].split(':')[1];
+            client['hostname_port'] =  hostname + ':' + port;
+
+            return client;
+        });
 
-            var port = remote_address.split(":").pop()
```

Comment:
I used .pop() intentionally here so this handles ipv6 correctly (in theory at least?)